test/unit/extension/lib/mergeCartsOnLoginTest.js

```diff
@@ -111,19 +111,43 @@ describe('mergeCartsOnLogin', () => {
       }
     }
 
-    const pipelineStub = sandbox.createStubInstance(ShopgateCartPipeline)
-    sandbox.stub(ShopgateCartPipeline, 'createForDevice').returns(pipelineStub)
-    sandbox.stub(ShopgateCartPipeline, 'create').returns(pipelineStub)
-    pipelineStub.get.returns(getCartResponse)
+    const anonymousPipelineStub = sandbox.createStubInstance(ShopgateCartPipeline)
+    const loggedInPipelineStub = sandbox.createStubInstance(ShopgateCartPipeline)
+    sandbox.stub(ShopgateCartPipeline, 'createForDevice').returns(anonymousPipelineStub)
+    sandbox.stub(ShopgateCartPipeline, 'createForUser').returns(loggedInPipelineStub)
+
+    anonymousPipelineStub.get.returns(getCartResponse)
+
+    anonymousPipelineStub.getCartId.returns('my_cart_id')
+    loggedInPipelineStub.getCartId.returns('user_cart_id')
 
     await mergeCartsOnLogin(context, {})
 
-    sinon.assert.calledWith(pipelineStub.addProducts, [
+    sinon.assert.calledWith(loggedInPipelineStub.addProducts, [
       {productId: 120, quantity: 1},
       {productId: 121, quantity: 1}
```

Comment:
Even though a unit test should test just one thing - you could change the qty of one of those tests products to 2

Reply:
I'm not sure it makes much difference as this test can only test input parameters not the merge itself. I did the change however.